### PR TITLE
Unskipped all gallery examples

### PR DESCRIPTION
**Pull request by nabobalis**
I unskipped all the examples, they all seem to work fine. Not sure why we skipped them.

Found a bug in the noaa client. 

Also, tidied up the tox.ini and setup.cfg file for our coverage to go back to more normal.

### CODE REVIEW

File: sunpy/net/dataretriever/sources/noaa.py [hunk 1]
```diff
@@ -127,7 +121,7 @@ def _get_url_for_timerange(self, timerange, **kwargs):
             return SRSClient._get_default_uri()
         result = list()
         base_url = 'ftp://ftp.swpc.noaa.gov/pub/warehouse/'
-        total_days = int(timerange.days) + 1
+        total_days = int(np.round(timerange.days.value) + 1)
```
Comment (nabobalis) [inline sunpy/net/dataretriever/sources/noaa.py]: I am not sure this is what I want to do. 

 In the example that triggered this, the `days.value` was `0.9`, so `int(0.9) + 1` is one day. But this has made it `round + 1`.

File: sunpy/net/dataretriever/sources/noaa.py [hunk 2]
```diff
@@ -127,7 +119,7 @@ def _get_url_for_timerange(self, timerange, **kwargs):
             return SRSClient._get_default_uri()
         result = list()
         base_url = 'ftp://ftp.swpc.noaa.gov/pub/warehouse/'
-        total_days = int(timerange.days) + 1
+        total_days = int(timerange.days.value) + 1
```
Comment (nabobalis) [inline sunpy/net/dataretriever/sources/noaa.py]: We apparently didn't trigger this until I enabled one of the examples.

File: tox.ini [hunk 3]
```diff
@@ -18,8 +18,8 @@ deps =
     online: pytest-rerunfailures
     online: pytest-timeout
 commands =
-    offline,astropydev,numpydev: pytest --verbose --pyargs sunpy -m "not figure" --cov=sunpy --rootdir={toxinidir} {posargs} {toxinidir}/docs
-    online: pytest --verbose --pyargs sunpy --timeout=300 -m "not figure" --remote-data=any --cov=sunpy  --rootdir={toxinidir} {posargs} {toxinidir}/docs
+    offline,astropydev,numpydev: pytest --verbose --pyargs sunpy -m "not figure" --cov=sunpy --rootdir={toxinidir} --cov-config={toxinidir}/setup.cfg {posargs} {toxinidir}/docs
+    online: pytest --verbose --pyargs sunpy --timeout=300 -m "not figure" --remote-data=any --cov=sunpy --rootdir={toxinidir} --cov-config={toxinidir}/setup.cfg {posargs} {toxinidir}/docs
```
Comment (nabobalis) [inline tox.ini]: I need to figure out some way to define a base command. These are getting ugly and out of hand.

File: setup.cfg [hunk 4]
```diff
@@ -98,34 +98,30 @@ known_first_party = sunpy
...
-   sunpy/data/_sample.py
+   */sunpy/conftest.py
+   */sunpy/version*
+   */sunpy/cython_version*
+   */sunpy/*setup*
+   */sunpy/extern/*
+   */sunpy/version*
+   */sunpy/__init__*
+   */sunpy/data/sample.py
+   */sunpy/data/_sample.py
```
Comment (nabobalis) [inline setup.cfg]: This seems to work for our sdist tests now.

**Comment by nabobalis** [comment]
Timeout on a test seems to be the issue.

File: setup.cfg [hunk 5]
```diff
@@ -1,6 +1,6 @@
 [metadata]
 name = sunpy
-package_name = sunpy # astropy compat
```
Comment (Cadair) [inline setup.cfg]: This comment is here to indicate we should remove it at somepoint as it is not specced by setuptools, but used by some astropy tooling like `sphinx-astropy`

Comment (nabobalis) [reply setup.cfg]: But we can just leave it there forever.

Comment (nabobalis) [reply setup.cfg]: WE WILL DELETE IT WHEN THE TIME COMES

**Comment by nabobalis** [comment]
FOREVER HAUNTED BY `ValueError: No operational GOES satellites on 1983-05-01 00:00:00 `

File: setup.cfg [hunk 6]
```diff
@@ -57,9 +56,9 @@ sunpy.tests = figure_hashes_py36.json
 
 [options.entry_points]
 asdf_extensions =
-    sunpy = sunpy.io.special.asdf.extension:SunpyExtension
-pytest11 =
-    asdf = asdf.tests.schema_tester
+  sunpy = sunpy.io.special.asdf.extension:SunpyExtension
+#pytest11 =
+#  asdf = asdf.tests.schema_tester
```
Comment (nabobalis) [inline setup.cfg]: This was the reason. I am not sure how we work around this.

**Comment by nabobalis** [comment]
Something is still wrong: 
Screenshot from 2019-03-11 23-26-11

**Comment by nabobalis** [comment]
Something else is wrong:
Screenshot from 2019-03-11 23-30-02

File: .circleci/config.yml [hunk 7]
```diff
@@ -6,6 +6,7 @@ merge-check: &merge-check
   name: Check if we need to merge upstream master
   command: |
     if [[ -n "${CIRCLE_PR_NUMBER}" ]]; then
+      git fetch origin --tags
```
Comment (nabobalis) [inline .circleci/config.yml]: This did not work.

**Pull request merged**